test-fork/AddSameLpTokenTwice.spec.ts

```diff
@@ -133,93 +132,111 @@ describe("Add same LP Token twice", () => {
-                expect(fromPool.lptoken, "lptokens").to.be.eq(toPool.lptoken);
-                expect(fromPid, "pid").to.not.eq(toPid);
-                expect(fromCrvRewardsBefore, "fromCrvRewards balance before").gt(0);
-                // extraRewards
-                await fromCrvRewards.approve(gaugeMigrator.address, fromCrvRewardsBefore);
-
-                // When it migrates it's full position
-                await gaugeMigrator.connect(account.signer).migrate(fromPid, toPid);
-
-                // Then previous lp position must be zero and new position must hold all its previous liquidity
-                const fromCrvRewardsAfter = await fromCrvRewards.balanceOf(account.address);
-                const toCrvRewardsAfter = await toCrvRewards.balanceOf(account.address);
-
-                expect(fromCrvRewardsAfter, "from crvRewards balance").to.be.eq(0);
-                expect(toCrvRewardsAfter, "to crvRewards balance").to.be.eq(
-                    toCrvRewardsBefore.add(fromCrvRewardsBefore),
-                );
-            };
+            oldCrvRewards = BaseRewardPool4626__factory.connect(poolInfo.crvRewards, lpWhale.signer);
+        });
+        it("deposit lp tokens to old pool", async () => {
+            const balance = await lpToken.balanceOf(lpWhale.address);
+            const amount = simpleToExactAmount(100);
+            expect(balance).to.gt(amount);
+
+            const crvRewardsBefore = await oldCrvRewards.balanceOf(lpWhale.address);
 
-            it("deploy GaugeMigrator", async () => {
-                gaugeMigrator = await deployContract<GaugeMigrator>(
+            expect(amount).gt(0);
+            expect(crvRewardsBefore, "crvRewards balance before").eq(0);
+            await lpToken.approve(phase2.booster.address, amount);
+            const stake = true;
+
+            await phase2.booster.connect(lpWhale.signer).deposit(oldPid, amount, stake);
+
+            const depositTokenBalance = await oldCrvRewards.balanceOf(lpWhale.address);
+            expect(depositTokenBalance, "crvRewards balance").eq(amount);
+        });
+        context("add new pool", async () => {
+            it("mock gauge controller", async () => {
+                // Mock gauge controller with one that returns 1 when you query the weight
+                await network.provider.send("hardhat_setCode", [
+                    config.addresses.gaugeController,
+                    MockGaugeController__factory.bytecode,
+                ]);
+                await network.provider.send("hardhat_setCode", [gauge, MockCurveGauge__factory.bytecode]);
+
+                // Send some tokens to voter proxy to avoid calling the gauge while withdrawing, the curve gauge mock has an issue.
+                const voterProxy = "0xaf52695e1bb01a16d33d7194c28c42b10e0dbec2";
+                await lpToken.transfer(voterProxy, simpleToExactAmount(200));
```

Comment:
I think we should try and get this test working without this. Can't see anything obvious as to why the mocked gauge controller is failing when calling gauge.withdraw. We should dig around until we find a solution 💪

Reply:
or if setting the gauge contract to be a mock, it should be done before the original deposit is made, as there is no token balance on that address since the deposit is made before. Would need to deploy it and actually add the lptoken to storage tho

Reply:
Impersonating whoever can add gauges to the gauge controller probably best